### DISCUSSION TITLE
How to disable styles?

### DISCUSSION THREAD

**Discussion by bulletmark** [hamdanal/rich-argparse#120] [Q&A]
I have a number of Python CLI programs, some of which use subparsers, so I thought I'd try this to see how it looks. However, I find that it interprets rich styles in my code which I did not intend. E.g. it seems I have commonly written "applications[s]" to indicate one or more applications but I am getting a strike-through on all those lines. How can I disable style interpretation?

I already tried `RichHelpFormatter.highlights.clear()` and `RichHelpFormatter.usage_markup = False` but no difference.

**Comment by hamdanal** [comment]
I am afraid this cannot be disabled "globally" at the moment. rich-argparse considers the description and help texts as [console markup](https://rich.readthedocs.io/en/latest/style.html) to allow for a simple usage like this:
```python
parser.add_argument("--foo", help="this argument is very [bold]IMPORTANT:[/bold]")
parser.add_argument("--bar", help="this is a [red]DANGEROUS[/red] argument, use it carefully")
```

You can escape these brackets though using a backslash and rich will know that they should not be interpreted as markup. In your example you'd write `"applications\[s\]"`.

With that being said, I can see that this may be considered cumbersome for a large application with lots of these escapes. I'll look into adding an option to disable markup globally.

**Comment by bulletmark** [comment]
I was aware what this problem was caused by and of course I know I can go through all my help/usage message strings and escape them but I am simply surprised that Rich style markup is interpreted by default. The README here says when porting from vanilla `argparse` to simply change `formatter_class=RichHelpFormatter` but it should probably be pointed out that you should also check for inadvertent markup across all your usage strings.

This also makes it awkward to fall-back to normal `argparse` at runtime, e.g. if `rich-argparse` is not installed, because `argparse` shows the escape chars.

**Comment by hamdanal** [comment]
I've add two new options to disable markup.
You can now set `RichHelpFormatter.help_markup = False` to disable console markup in the help text of arguments and `RichHelpFormatter.text_markup = False` to disable console markup in the descriptions and epilog.
I also added a section about this in the README.

This will be included in the next release which will be out soon. Thank you for the feedback.
